engine/access/rpc/backend/backend_scripts.go

```diff
@@ -90,18 +93,27 @@ func (b *backendScripts) executeScriptOnExecutionNode(
 	if err != nil {
 		return nil, status.Errorf(codes.Internal, "failed to find execution nodes at blockId %v: %v", blockID.String(), err)
 	}
+	// encode to MD5 as low compute/memory lookup key
+	encodedScript := md5.Sum(script) //nolint:gosec
+	tenMinutes := time.Duration(10) * time.Minute
 
 	// try each of the execution nodes found
 	var errors *multierror.Error
 	// try to execute the script on one of the execution nodes
 	for _, execNode := range execNodes {
+		executionTime := time.Now()
 		result, err := b.tryExecuteScript(ctx, execNode, execReq)
 		if err == nil {
-			b.log.Debug().
-				Str("execution_node", execNode.String()).
-				Hex("block_id", blockID[:]).
-				Str("script", string(script)).
-				Msg("Successfully executed script")
+			timestamp, seen := b.seenScripts[encodedScript]
+			// log unique script seen in the last 10 minutes
+			if !seen || executionTime.Sub(timestamp) >= tenMinutes {
+				b.log.Debug().
+					Str("execution_node", execNode.String()).
+					Hex("block_id", blockID[:]).
+					Str("script", string(script)).
+					Msg("Successfully executed script")
+				b.seenScripts[encodedScript] = executionTime
```

Comment:
so if we don't have an LRU, wouldn't this map grow indefinitely? There can be a large number of different type of scripts

Reply:
yes! but since the script is compressed to a 16 byte identifier, and I assumed there aren't an order of ~1,000,000 unique scripts, so this wouldn't be an issue. I can switch to LRU, what's a reasonable cache size?

Reply:
tbh - i don't know what the cache size should be but we can start with ~10K maybe?

Reply:
btw - is there a way put things in the cache only if we are logging at debug level? Then we could still run with just a map since in production we run with log level `info`. Only when we want to see the scripts we swtich to log level debug.

Reply:
Also - @yhassanzadeh13 has a new HeroCache which is an optimized LRU. So lets use that if we are switching.

Reply:
a way to cache only when we're logging at debug level is to just put in a check like `if log.level == log.DebugLevel`. Looked up HeroCache, it's meant for `flow.Identifier` -> `flow.Entity` key/vals for which `script string -> flow.Entity` casts have to be done. I'm leaning on using the `if` branch for space/computation simplicity.

Reply:
thanks for looking into HeroCache. 
Ok yeah lets go with `if` branch.